apps/web/src/components/(dashboard)/common/command-menu.tsx

```diff
@@ -0,0 +1,112 @@
+'use client';
+
+import { useState } from 'react';
+
+import { useRouter } from 'next/navigation';
+
+import { Monitor, Moon, Sun } from 'lucide-react';
+import { useTheme } from 'next-themes';
+import { useHotkeys } from 'react-hotkeys-hook';
+
+import {
+  CommandDialog,
+  CommandEmpty,
+  CommandGroup,
+  CommandInput,
+  CommandItem,
+  CommandList,
+  CommandShortcut,
+} from '@documenso/ui/primitives/command';
+
+export function CommandMenu() {
+  const { setTheme } = useTheme();
+  const { push } = useRouter();
+  const [open, setOpen] = useState(false);
+  const [search, setSearch] = useState('');
+  const [pages, setPages] = useState<string[]>([]);
+  const page = pages[pages.length - 1];
+
+  useHotkeys('ctrl+k', (e) => {
+    e.preventDefault();
+    setOpen((open) => !open);
+  });
+  useHotkeys('n+s', () => push('/settings'));
+  useHotkeys('n+d', () => push('/documents?status=ALL'));
+
+  return (
+    <CommandDialog
+      commandProps={{
+        onKeyDown: (e) => {
+          // Escape goes to previous page
+          // Backspace goes to previous page when search is empty
+          if (e.key === 'Escape' || (e.key === 'Backspace' && !search)) {
+            e.preventDefault();
+            setPages((pages) => pages.slice(0, -1));
+          }
+        },
+      }}
+      open={open}
+      onOpenChange={setOpen}
+    >
+      <CommandInput
+        value={search}
+        onValueChange={setSearch}
+        placeholder="Type a command or search..."
+      />
+      <CommandList>
+        <CommandEmpty>No results found.</CommandEmpty>
+        {!page && (
```

Comment:
This is pretty cool, we should be mindful to split these out in the future as we add more.